datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -177,7 +179,7 @@ impl LogicalPlanBuilder {
     /// so it's usually better to override the default names with a table alias list.
     ///
     /// If the values include params/binders such as $1, $2, $3, etc, then the `param_data_types` should be provided.
-    pub fn values(mut values: Vec<Vec<Expr>>) -> Result<Self> {
+    pub fn values(values: Vec<Vec<Expr>>, schema: Option<&DFSchemaRef>) -> Result<Self> {
```

Comment:
👍  This makes a lot of sense to me to pass in the schema too if knon. 

I think it might be nicer on users if we didn't make an API change and instead added a new API like 

```rust
pub fn values_with_schema(values: Vec<Vec<Expr>>, schema: Option<&DFSchemaRef>) -> Result<Self> {
...
}
```

Even if we also deprecated `values` it would help users prepare for upgrade